sigstore-java/src/test/java/dev/sigstore/timestamp/client/TimestampVerifierTest.java

```
@@ -0,0 +1,258 @@
+package dev.sigstore.timestamp.client;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.google.common.io.Resources;
+import com.google.protobuf.util.JsonFormat;
+import dev.sigstore.proto.trustroot.v1.TrustedRoot;
+import dev.sigstore.trustroot.SigstoreTrustedRoot;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.spec.InvalidKeySpecException;
+import org.bouncycastle.asn1.cmp.PKIFailureInfo;
+import org.bouncycastle.asn1.cmp.PKIStatus;
+import org.bouncycastle.tsp.TSPException;
+import org.bouncycastle.tsp.TimeStampResponseGenerator;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class TimestampVerifierTest {
+  private static SigstoreTrustedRoot trustedRoot;
+  private static SigstoreTrustedRoot trustedRootWithOutdatedTsa;
+  private static SigstoreTrustedRoot trustedRootWithMultipleTsas;
+  private static byte[] trustedTsRespBytesWithEmbeddedCerts;
+  private static byte[] trustedTsRespBytesWithoutEmbeddedCerts;
+  private static byte[] invalidTsRespBytes;
+  private static byte[] untrustedTsRespBytes;
+
+  @BeforeAll
+  public static void loadResources() throws IOException {
+    // Response from Sigstore TSA (in trusted root) with embedded certs
+    try (var is =
+        Resources.getResource(
+                "dev/sigstore/samples/timestamp-response/valid/sigstore_tsa_response_with_embedded_certs.tsr")
+            .openStream()) {
+      if (is == null) {
+        throw new IOException(
+            "dev/sigstore/samples/timestamp-response/valid/sigstore_tsa_response_with_embedded_certs.tsr");
+      }
```

Handling `is == null` is maybe too cautious, just let the init fail in tests.